File: modules/webrtc-core/src/main/java/com/shterneregen/securelan/webrtc/event/RtcStateChangedEvent.java
package com.shterneregen.securelan.webrtc.event;

import com.shterneregen.securelan.common.model.rtc.RtcSessionMode;
import com.shterneregen.securelan.common.model.rtc.RtcSessionState;

public record RtcStateChangedEvent(
        String sessionId,
        String remotePeer,
        RtcSessionMode mode,
        RtcSessionState state,
        String message
) implements RtcEvent {
}

File: modules/webrtc-core/src/main/kotlin/com/shterneregen/securelan/webrtc/event/RtcStateChangedEvent.kt
package com.shterneregen.securelan.webrtc.event

import com.shterneregen.securelan.common.model.rtc.RtcSessionMode
import com.shterneregen.securelan.common.model.rtc.RtcSessionState

@JvmRecord
data class RtcStateChangedEvent(
    val sessionId: String?,
    val remotePeer: String?,
    val mode: RtcSessionMode?,
    val state: RtcSessionState?,
    val message: String?,
) : RtcEvent
